### PR TITLE
Show token usage and formatting of code

### PR DESCRIPTION
**Pull request by Dev-Khant**
Resolves https://github.com/langfuse/langfuse/issues/190

I think this is the necessary info for users. Let me know if we want to add more. Thanks :)

### CODE REVIEW

File: langfuse/callback.py [hunk 1]
```diff
@@ -524,7 +538,9 @@ def on_llm_end(
                 )
 
                 self.runs[run_id].state = self.runs[run_id].state.update(
-                    UpdateGeneration(completion=extracted_response, end_time=datetime.now(), usage=llm_usage)
+                    UpdateGeneration(
+                        completion=extracted_response, end_time=datetime.now(), usage=llm_usage, metadata=token_metadata
```
Comment (maxdeichmann) [inline langfuse/callback.py:542]: Why do you want to put the token metadata on the metadata as well? We are already tracking that in `usage`.

Comment (Dev-Khant) [reply langfuse/callback.py:542]: I thought of giving all the information in metadata. But will remove it because in `usage` it's included.

**Comment by Dev-Khant** [comment]
@maxdeichmann Here will we get the `headers` from the response because base class `LLMResult` does not contain headers? So we have to find a way to get headers in `on_llm_end`

File: langfuse/callback.py (hunk 1 above)
Comment (maxdeichmann) [reply langfuse/callback.py:542]: Awesome

**Comment by maxdeichmann** [comment]
> @maxdeichmann Here will we get the `headers` from the response because base class `LLMResult` does not contain headers? So we have to find a way to get headers in `on_llm_end`

Yes, i am not sure whether and if yes, how Langchain exposes these headers. Did you print out all the kwargs and the LLMResult? They might be in there.

**Comment by Dev-Khant** [comment]
Here I think they are using `chatcompletion` method because in LLMResults they have no header info.
And all the info are only generated in  `chatcompletion`
image

**Comment by Dev-Khant** [comment]
Here I think they are using `chatcompletion` method because in LLMResults they have no header info.
And all the info are only generated in  `chatcompletion`
image

**Comment by maxdeichmann** [comment]
@Dev-Khant thanks a lot for the contribution. I hope it is fine for you if i close this as we are not able to get more header context from langchain unfortunately. Again, thanks a lot!

**Pull request closed without merging**